src/NServiceBus.Core/Persistence/InMemory/Gateway/InMemoryGatewayPersistence.cs

```diff
@@ -15,6 +18,11 @@ internal InMemoryGatewayPersistence()
         /// </summary>
         protected internal override void Setup(FeatureConfigurationContext context)
         {
+            if (!PersistenceStartup.HasSupportFor<StorageType.GatewayDeduplication>(context.Settings))
```

Comment:
This is indeed vierd, but its a result of the core wrongly containing the persistence interface for the GW but the GW it self has been moved out of the core.

Long term we should:

1. Move the interface to the GW
2. Create separate NServiceBus.NHibernate.GatwayStorage and  NServiceBus.RavenDB.GatwayStorage packages
3. Move the inmemory deduplication to the GW

I propose we raise a different issue for this?